javac-plugin/src/test/java/com/aws/jverify/plugin/CompilationRunner.java

```diff
@@ -0,0 +1,123 @@
+package com.aws.jverify.plugin;
+
+import com.google.errorprone.annotations.Var;
+import com.google.testing.compile.Compilation;
+import com.sun.source.doctree.ValueTree;
+import org.jetbrains.annotations.NotNull;
+
+import javax.tools.JavaFileObject;
+import java.io.File;
+import java.io.IOException;
+import java.lang.reflect.Method;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class CompilationRunner {
+
+    /**
+     * Runs the main method of a generated class from a Compilation and returns the exit code.
+     * This method launches the class in a separate JVM process to properly capture the exit code.
+     */
+    public static int runGeneratedClassAndGetExitCode(
+            Compilation compilation, Iterable<File> classPath, String qualifiedClassName, String... args) {
+
+        Path tempDir;
+        try {
+            tempDir = Files.createTempDirectory("generated-classes");
+        } catch (IOException e) {
+            System.err.println("Failed to create temp directory: " + e.getMessage());
+            return -1;
+        }
+
+        try {
+            writeCompiledClassFiles(compilation, tempDir);
+
+            // Prepare command to run the class in a separate JVM
+            String javaHome = System.getProperty("java.home");
+            String javaBin = javaHome + File.separator + "bin" + File.separator + "java";
+
+            var process = getProcess(classPath, qualifiedClassName, args, javaBin, tempDir);
+
+            int exitCode = process.waitFor();
+            return exitCode;
+
+        } catch (Exception e) {
+            System.err.println("Error running generated class: " + e.getMessage());
+            e.printStackTrace();
+            return -1;
+        } finally {
+            // Clean up temporary directory
+            try {
+                deleteRecursively(tempDir);
+            } catch (IOException e) {
+                System.err.println("Warning: Failed to delete temp directory: " + e.getMessage());
+            }
+        }
+    }
+
+    @NotNull
+    private static Process getProcess(Iterable<File> classPath, String qualifiedClassName, String[] args, String javaBin, Path tempDir) throws IOException {
+        List<String> command = new ArrayList<>();
+        command.add(javaBin);
+        command.add("-cp");
+        StringBuilder arg = new StringBuilder();
+        for(var cpFile : classPath) {
+            arg.append(cpFile.toString()).append(":");
```

Comment:
Windows uses `;` instead of `:` to separate classpath entries - consider using:

```suggestion
            arg.append(cpFile.toString()).append(File.pathSeparator);
```